src/getCurrentConnectionInfo.ts

```diff
@@ -0,0 +1,32 @@
+export type ConnectionInfo = {
+  connectionIndex: number;
+
+  /** an optional string like "designer" | "explorer" | "studio" */
+  product?: string;
+};
+
+/** returns info on the current Portal/Cloud connection */
+export const getCurrentConnectionInfo = (
+  window: Window
+): ConnectionInfo | null => {
+  const urlPathRegex =
+    /^\/u\/(?<connectionIndex>[0-9]+)\/(?<productWithSlash>.+?\/)?/;
```

Comment:
You can simplify this regex to this:

```
const urlPathRegex = /^\/u\/(?<connectionIndex>[0-9]+)\/(?<product>[^/]+)?/;
```

<img width="827" alt="Screen Shot 2022-03-02 at 4 51 18 PM" src="https://user-images.githubusercontent.com/518026/156463123-10199d0a-26a8-465d-9e6f-e1d86a25ac36.png">

Then you don't need to remove the slash instead just include all chars except `/` you could also change this to `/(?<product>[^/]*)/` which will match 0 or more non-slashes, then product will be an empty string

Reply:
[`65fc956` (#8)](https://github.com/stardog-union/portal-sdk/pull/8/commits/65fc95676eef30aff1fa4ee976acb2a28be37ad6)